org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/TraceRegionMerger.java

```diff
@@ -177,7 +179,7 @@ private AbstractTraceRegion replaceTruncated(int index, AbstractTraceRegion regi
 	}
 
 	private AbstractTraceRegion replaceMerged(int index, AbstractTraceRegion region, List<ILocationData> locationData) {
-		List<ILocationData> mergedLocations = newArrayList(region.getAssociatedLocations());
+		Set<ILocationData> mergedLocations = newLinkedHashSet(region.getAssociatedLocations());
 		mergedLocations.addAll(locationData);
 		AbstractTraceRegion merged = new TraceRegion(
 				region.getMyOffset(), region.getMyLength(),
```

Comment:
True, but it has been around a bit and I know projects that use it. So I'd like to avoid breaking it if there is no good reason. OTOH copying twice is indeed not a good idea (I am unsure if the copying makes sense at all). If you prefer your proposed solution, could you add an overloaded constructor with the old signature and deprecate it?